File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.cpp
class Solution {
    struct compare {
        bool operator()(pair<int,int> a, pair<int,int> b) {
            return a.first > b.first;
        }
    };

public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        priority_queue<pair<int, int>, vector<pair<int,int>>, compare> minheap;
        unordered_map<int,int> freq;
        for(int i =0 ; i<nums.size(); i++) {
            freq[nums[i]] ++;
        }
        for(auto p: freq) {
            minheap.push({p.second, p.first});
            if(minheap.size() > k) {
                minheap.pop();
            }
        }
        vector<int> result;
        while(k-- && !minheap.empty()) {
            auto p = minheap.top();
            minheap.pop();
            result.push_back(p.second);
        }
        return result;
    }
};

File: Data Structures & Algorithms/kth-largest-element-in-an-array/submission-1.java
class Solution {
    public int findKthLargest(int[] nums, int k) {
        PriorityQueue<Integer> pq = new PriorityQueue<Integer>((a, b) -> Integer.compare(a, b));
        for(int i : nums) {
            pq.add(i);
            System.out.println("number added"+ i + " top of the q = "+ pq.peek());
            if(pq.size() > k){ pq.poll();}
        }
        return pq.isEmpty() ? 0 : pq.peek();
    }
}
